fe/fe-core/src/main/java/org/apache/doris/nereids/StatementContext.java

```diff
@@ -508,6 +515,67 @@ public void addPlannerHook(PlannerHook plannerHook) {
         this.plannerHooks.add(plannerHook);
     }
 
+    /**
+     * refTables
+     *
+     * @param tables tables
+     */
+    public void refTables(Map<List<String>, TableIf> tables) {
+        if (tables == null) {
+            return;
+        }
+        for (TableIf tableIf : tables.values()) {
+            if (tableIf instanceof MvccTable) {
+                MvccTable mvccTable = (MvccTable) tableIf;
+                long snapshotId = fetchSnapshotIdIfNotExist(mvccTable);
+                mvccTable.ref(snapshotId);
+            }
+        }
+    }
+
+    /**
+     * unrefTables
+     */
+    public void unrefTables() {
+        for (Entry<BaseTableInfo, Long> entry : refSnapshotIds.entrySet()) {
```

Comment:
what will happen if we ref snapshot A here and table has been updated to snapshot B before load cache?